onnx/common/ir.h

```diff
@@ -1248,17 +1249,27 @@ inline const Graph* Value::owningGraph() const {
 // `captured` nodes in subgraph determines which value it captures
 // by storing the value's unique name, so old unique names in `captured` nodes
 // should also be updated.
-inline Value* Value::setUniqueName(const std::string& name, bool rename_subgraph_captured_nodes) {
-  if (has_unique_name() && rename_subgraph_captured_nodes) {
-    auto* graph = owningGraph();
-    graph->forEachNode([this, &name](Node* node) {
+// Initializer names are also storaged in graph.initializer_names_, it should be
+// updated too.
+inline Value* Value::setUniqueName(const std::string& name, bool update_related_names) {
+  if (has_unique_name() && update_related_names) {
+    auto old_name = unique_name_;
+    for (size_t i = 0; i < owningGraph()->initializer_names_.size(); i++) {
+      auto &initializer_name = owningGraph()->initializer_names_[i];
+      if (initializer_name == old_name) {
+        initializer_name = name;
+        owningGraph()->initializers_[i].setName(name);
+      }
+    }
+    auto *graph = owningGraph();
```

Comment:
tiny nit: could we declare `auto *graph = owningGraph();` in advance before L1257 for later use (L1257-L1261)? Or perhaps directly use `owningGraph()` all the time without having a variable.

Reply:
sorry for the late response~ updated